### PR TITLE
Onboarding 

### PR DESCRIPTION
**Pull request by boek**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -74,6 +75,7 @@ class HomeFragment : Fragment(), CoroutineScope {
     private var homeMenu: HomeMenu? = null
 
     var deleteSessionJob: (suspend () -> Unit)? = null
+    private var layoutManagerSate: Parcelable? = null
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: nit - layoutManagerState*

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 2]
```diff
@@ -180,6 +185,26 @@ class HomeFragment : Fragment(), CoroutineScope {
...
     }
 
+    override fun onViewStateRestored(savedInstanceState: Bundle?) {
+        savedInstanceState?.apply {
+            layoutManagerSate = getParcelable(KEY_LAYOUT_MANAGER_STATE)
+            val progress = getFloat(KEY_MOTION_LAYOUT_PROGRESS)
+            homeLayout.progress = if (progress > MOTION_LAYOUT_PROGRESS_ROUND_POINT) 1.0f else 0f
+        }
+
+        super.onViewStateRestored(savedInstanceState)
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: nit - I'm not sure if it matters or works the same way here but most examples I've seen have `super.onViewStateRestored(savedInstanceState)` first

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 3]
```diff
@@ -180,6 +185,26 @@ class HomeFragment : Fragment(), CoroutineScope {
...
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        sessionControlComponent.view.layoutManager!!.onSaveInstanceState()!!.apply {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: same here super.onSaveInstanceState(outState) maybe should be first if it still works?

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingTrackingProtectionViewHolder.kt [hunk 4]
```diff
@@ -7,17 +7,49 @@ package org.mozilla.fenix.home.sessioncontrol.viewholders.onboarding
...
+            updateTrackingProtectionSetting(isChecked)
+        }
     }
+
+    private fun updateTrackingProtectionSetting(enabled: Boolean) {
+        Settings.getInstance(view.context).setTrackingProtection(enabled)
+        with(view.context.components.core) {
+            val policy = createTrackingProtectionPolicy(enabled)
+            engine.settings.trackingProtectionPolicy = policy
+            sessionManager.sessions.forEach {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingTrackingProtectionViewHolder.kt]: There's a new use case for this so we don't have to go through each session see: https://github.com/mozilla-mobile/fenix/commit/cfca683a14ddc06bd09ea320bb5710cdcf001c80#diff-419111675b2c7ef0fa1219710d528e60R37

**Comment by boek** [comment]
The Firefox Accounts onboarding card has a darker style (hence: dark) that is the same across every theme. Where the light cards are visually lighter, though these get styled appropriately in the themes.

**Pull request merged**